### PR TITLE
Adding solutions to instructor notes

### PR DESCRIPTION
**Pull request by PBarmby**
Addresses #85. Still a work in progress, so please don't merge yet. Just looking for confirmation that solutions here seem reasonable as far as formatting and content.

### CODE REVIEW

**Comment by PBarmby** [comment]
@tbekolay @abostroem: This is ready for a look now -- I haven't done the solutions for the command-line section but won't have time to get to that for a while. I can fix small problems in the earlier sections, though, and have addressed all the comments above.

**Comment by abostroem** [comment]
@PBarmby Thank you so much for putting these together. I want to take some time to read them so I probably won't to commenting/merging for a few more weeks. If @tbekolay has time before then, maybe we can get it done sooner.

File: instructors.md [hunk 1]
```diff
@@ -90,18 +90,395 @@ which are covered in the lesson on the shell.
...
+Solutions to exercises:
+
+> ## From 1 to N {.challenge}
+> Using `range`,
+> write a loop that uses `range` to print the first 3 natural numbers:
+>
+> ~~~ {.python}
+> for i in range(1,4):
+>    print i
+> 1
```
Comment (abostroem) [inline instructors.md]: I'd like to put all outputs in the the output CSS class ~~~{.output}

File: instructors.md [hunk 2]
```diff
@@ -90,18 +90,395 @@ which are covered in the lesson on the shell.
...
+> ~~~ {.python}
+> newstring = ""
+> oldstring = "Newton"
+> for c in range(len(oldstring)-1,-1,-1):
+>    newstring = newstring + oldstring[c]
+> print result
+> "notweN"
+> ~~~
+
+After discussing these challenges could be a good time to introduce the `b*=2` syntax.
```
Comment (abostroem) [inline instructors.md]: Do we really want to recommend students be introduced to this syntax?

File: instructors.md [hunk 3]
```diff
@@ -90,18 +90,395 @@ which are covered in the lesson on the shell.
...
+> ~~~ {.python}
+> def outer(input_string):
+>     return input_string[0] + input_string[-1]
+> ~~~
+
+> ## Rescaling, with parameters {.challenge}
+>
+> Rewrite the `rescale` function so that it scales data to lie between 0.0 and 1.0 by default,
+> but will allow the caller to specify lower and upper bounds if they want.
+> FIXME: looks like the rescale function is no longer in the lesson? FIXME
```
Comment (abostroem) [inline instructors.md]: Writing the rescale function is the 3rd exercise

File: instructors.md [hunk 4]
```diff
@@ -90,18 +90,395 @@ which are covered in the lesson on the shell.
...
+> What pre-conditions and post-conditions would you write for it?
+> Compare your answer to your neighbor's:
+> can you think of a function that will pass your tests but not hers or vice versa?
+> Answer: 
+> possible pre-conditions: 
+> ~~~ {.python}
+> assert len(input)>0, 'List length must be non-zero'
+> import numbers, numpy
+> for n in input_list:
+>     assert isinstance(n, numbers.Number) and not numpy.isnan(n),  'list entries must be numbers'
```
Comment (abostroem) [inline instructors.md]: We haven't seen isinstance or isnan so I don't think we can expect students to come up with it in the exercises

File: instructors.md [hunk 5]
```diff
@@ -90,18 +90,395 @@ which are covered in the lesson on the shell.
...
+>     result = [values[0]]
+>     for v in values[1:]:
+>         assert result[-1] >= 0
+>         result.append(result[-1] + v)
+>         assert result[-1] >= result[0]
+>     return result
+> ~~~
+> Answer:
+> The first assertion checks that the input sequence `values` is not empty; an empty sequence such as `[]` will make it fail.
+> The second assertion checks that the last item in `result` exists (FIXME: not obvious to me how to make this fail)
```
Comment (abostroem) [inline instructors.md]: This is checking that you don't have negative numbers in your input list. For example this fails: 
values = range(-10, 1)
print running(values)

File: instructors.md [hunk 6]
```diff
@@ -90,18 +90,395 @@ which are covered in the lesson on the shell.
...
+>     for v in values[1:]:
+>         assert result[-1] >= 0
+>         result.append(result[-1] + v)
+>         assert result[-1] >= result[0]
+>     return result
+> ~~~
+> Answer:
+> The first assertion checks that the input sequence `values` is not empty; an empty sequence such as `[]` will make it fail.
+> The second assertion checks that the last item in `result` exists (FIXME: not obvious to me how to make this fail)  
+> The third assertion checks that the last item in `result` exists (FIXME: not obvious to me how to make this fail)
```
Comment (abostroem) [inline instructors.md]: This tests that your sum is always greater than the first value. For instance this fails:
values = range(5, 10)
values.append(-34)
print running(values)

**Comment by abostroem** [comment]
Thanks @PBarmby these are a great start!

**Comment by PBarmby** [comment]
Sorry to have taken so long in getting back to this. @abostroem, is it useful for me to try and finish this by making the changes you suggested above, or does the switch to Python 3 mean I should start over? Which branch should I submit the PR against?

**Comment by abostroem** [comment]
@PBarmby Please continue to work on this and submit the PR to the gh-pages branch. Once we have merged it we can update it to Python 3. It should mostly affect the print statements. How does that sound @tbekolay?

I merged a number of outstanding PR. I think it was mostly modifications on to the wording of some of the exercises - but please make sure your solution match the current set of exercises. 

Thank you so much for working on this - it is so important for instructors

**Comment by PBarmby** [comment]
Closing this as it's replaced by #144.

**Pull request closed without merging**